src/Lottery.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * A module to alert users if they have won a regularly running lottery.
+ *
+ * ## User Experience
+ *  * On a daily basis, this module will call an external API to check 
+ *    if a user is among the winners of a weekly lottery draw.
+ *      * If they are a winner and they have not yet redeemed their code, 
+ *        a redemption code will be pulled.
+ *      * If they are a winner, but they have already redeemed their code, 
+ *        nothing will be returned.
+ *      * If they are not a winner, nothing will be returned.
+ *  * If the user is a winner and they have not yet redeemed their code, 
+ *    a "Notification" will pop up on the right side of the screen and the
+ *    "browserAction" for the Beyond the Rally extension will be replaced.
+ *    The new "browserAction" icon will be a dollar sign and the resulting
+ *    pop-up for this will feature the redemption code, instructions on
+ *    how to redeem, and a large button labeled "Redeemed".
+ *  * When the user clicks the "Redeemed" button to indicate that the
+ *    redemption code has been redeemed, two events happen:
+ *      * A different external API is called that indicates to our "database"
+ *        that the code has been redeemed by the participant
+ *      * A listener is alerted that the code has been redeemed, which turns off
+ *        the "Notification" and "browserAction" for the day, returning things
+ *        to normal.
+ * 
+ * Mason Jiang, 2/2/2022
+ */
+
+import * as webScience from "@mozilla/web-science";
+ 
+let initialized = false
+
+/**
+ * Fetch lottery results from API
+ * @param {string} userID - The ID of the user.
+ */
+async function fetchResults( userID ) {
+  const lotteryurl = "https://oz6ke1zh2h.execute-api.us-west-2.amazonaws.com/default/gjmartin-rally"
+  const response = await fetch(lotteryurl, { // fetch results from API
+    method: 'POST', //
+    body: JSON.stringify({"userID":userID})
+  });
+
+  if (!response.ok) { // there was an error
```

Comment:
You could just `throw` here, it's an async function so that'll end up being a rejected promise to the caller.

Reply:
If you'd like to leave as-is, please make it `console.error` instead of `console.log`

Reply:
Will change to `console.error`